amqphandler/amqphandler.go

```diff
@@ -778,3 +759,53 @@ func (handler *AmqpHandler) notifyCloudDisconnected() {
 		consumer.CloudDisconnected()
 	}
 }
+
+func (handler *AmqpHandler) scheduleMessage(messageType string, data interface{}, important bool) error {
+	if !important && !handler.isConnected {
```

Comment:
use read mutex for isConnected var

Reply:
it is guarded with interface mutex. I don't see why it should be guarded with additional read mutex.